workflow/controller/operator.go

```diff
@@ -3517,7 +3517,18 @@ func (woc *wfOperationCtx) addFinalizers() {
 }
 
 func (woc *wfOperationCtx) addArtifactGCFinalizer() {
-	if woc.execWf.Spec.ArtifactGC != nil && woc.execWf.Spec.ArtifactGC.Strategy != wfv1.ArtifactGCNever {
+
+	artifactDefinedGC := false
+	for _, template := range woc.execWf.Spec.Templates {
```

Comment:
what about encapsulating into `Workflow.HasArtifactGC`?

Reply:
Yeah, thought about doing that myself. Will do.

Reply:
Done. Let me know if you think I should also create a method on the Template level or not.